src/libxmljs.cc

```diff
@@ -97,9 +98,53 @@ char* xmlMemoryStrdupWrap(const char* str)
     return res;
 }
 
-// callback function for `xmlDeregisterNodeDefault`
+void deregisterNsList(xmlNs* ns)
+{
+    while (ns != NULL) {
+        if (ns->_private != NULL) {
+            XmlNamespace* wrapper = static_cast<XmlNamespace*>(ns->_private);
+            wrapper->xml_obj = NULL;
+            ns->_private = NULL;
+        }
+        ns = ns->next;
+    }
+}
+
+void deregisterNodeNamespaces(xmlNode* xml_obj)
+{
+    xmlNs *ns = NULL;
```

Comment:
@defunctzombie – Funny! I actually changed my commit, after the initial push, from using `type *var` to `type* var`, because I saw the latter form is used exclusively in this file, and because I assumed my preference for the former style is a hangover from plain old C. I missed one usage: That first variable declared in `deregisterNodeNamespaces`.

I pushed a fix so that line is consistent, and now all pointer declarations in the file use asterisk next to type, rather than next to the variable name. I'm content to follow other convention, but I note that if I rewrite the commit to use `type *var`, they will no longer match the declarations in the rest of the file.

I didn't know about `clang-format`. Looks like a good thing.